src/olympia/api/tests/test_authentication.py

```diff
@@ -221,3 +221,8 @@ def test_verify(self):
         assert response.status_code == 200
         data = json.loads(response.content)
         assert data['token'] == token
+
+    def test_invalid_user_id(self):
+        token = self.client.generate_api_token(self.user, user_id=-1)
+        with self.assertRaises(AuthenticationFailed):
+            self._authenticate(token)
```

Comment:
I was thinking that it might be nice to add a test that authenticates, changes the user name, and re-authenticates to make sure authentication doesn't break. I thought of it is because the DRF-JWT lib has so much indirection that they could pull the rug out from under us quite easily. If that test becomes too hard to write then skip it but if it's easy I say add it.